tests/test_fields.py

```diff
@@ -140,6 +140,55 @@ def test_can_validate_german_decimals(self):
         self.assertEqual(decimal.Decimal("12345678"), german_decimal_field_format.validated("12.345.678"))
         self.assertEqual(decimal.Decimal("171234567.89"), german_decimal_field_format.validated("171.234.567,89"))
 
+    def test_can_set_rule_for_field_format(self):
+        field_format = fields.DecimalFieldFormat("x", False, None, "3.2...4.2", _ANY_FORMAT)
+        self.assertEqual(decimal.Decimal('3.2'), field_format.validated('3.2'))
+        self.assertEqual(decimal.Decimal('4.2'), field_format.validated('4.2'))
+
+        field_format = fields.DecimalFieldFormat("x", False, None, "3.2...", _ANY_FORMAT)
+        self.assertEqual(decimal.Decimal('3.2'), field_format.validated('3.2'))
+
+        field_format = fields.DecimalFieldFormat("x", False, None, "...4.2", _ANY_FORMAT)
+        self.assertEqual(decimal.Decimal('4.2'), field_format.validated('4.2'))
+
+    def test_can_set_length_for_field_format(self):
+        field_format = fields.DecimalFieldFormat("x", False, "1.1", "", _ANY_FORMAT)
+        self.assertEqual(decimal.Decimal('0'), field_format.validated('0'))
+        self.assertEqual(decimal.Decimal('-9.9'), field_format.validated('-9.9'))
+        self.assertEqual(decimal.Decimal('9.9'), field_format.validated('9.9'))
+        self.assertEqual(decimal.Decimal('.9'), field_format.validated('.9'))
+        self.assertEqual(decimal.Decimal('9.'), field_format.validated('9.'))
+
+    def test_fails_on_outer_range_for_field_length(self):
+        field_format = fields.DecimalFieldFormat("x", False, "1.1", "", _ANY_FORMAT)
+        self.assertRaises(errors.FieldValueError, field_format.validated, '10')
+        self.assertRaises(errors.FieldValueError, field_format.validated, '.01')
+
+    def test_fails_on_outer_range_for_field_rule(self):
+        field_format = fields.DecimalFieldFormat("x", False, None, "3.2...4.2", _ANY_FORMAT)
+        self.assertRaises(errors.FieldValueError, field_format.validated, '4.3')
+        self.assertRaises(errors.FieldValueError, field_format.validated, '3.1')
+
+        field_format = fields.DecimalFieldFormat("x", False, None, "3.2...", _ANY_FORMAT)
+        self.assertRaises(errors.FieldValueError, field_format.validated, '3.1')
+
+        field_format = fields.DecimalFieldFormat("x", False, None, "...4.2", _ANY_FORMAT)
+        self.assertRaises(errors.FieldValueError, field_format.validated, '4.3')
+
+    def test_fails_on_no_number(self):
```

Comment:
Rename to `test_fails_on_non_numeric_value` and remove redundant test `test_fails_on_broken_decimal`.